File: JAVA/GestionRecrutement/src/main/java/models/Interview.java
package models;

import java.time.LocalDateTime;
import java.util.Objects;

public class Interview {

    private int id;
    private int applicationId;
    private int interviewerId;
    private LocalDateTime interviewDate;
    private String type;
    private String location;
    private String meetingLink;
    private String feedback;
    private int score;
    private String result;
    private String candidateName; // Joined field
    private String interviewerName; // Joined field
    private String jobTitle; // Joined field

    public Interview() {
    }

    // Without id
    public Interview(int applicationId, int interviewerId,
            LocalDateTime interviewDate, String type,
            String location, String meetingLink,
            String feedback, int score, String result) {
        this.applicationId = applicationId;
        this.interviewerId = interviewerId;
        this.interviewDate = interviewDate;
        this.type = type;
        this.location = location;
        this.meetingLink = meetingLink;
        this.feedback = feedback;
        this.score = score;
        this.result = result;
    }

    // Full constructor
    public Interview(int id, int applicationId, int interviewerId,
            LocalDateTime interviewDate, String type,
            String location, String meetingLink,
            String feedback, int score, String result) {
        this(applicationId, interviewerId, interviewDate, type,
                location, meetingLink, feedback, score, result);
        this.id = id;
    }

    // Full constructor (including joined fields)
    public Interview(int id, int applicationId, int interviewerId,
            LocalDateTime interviewDate, String type,
            String location, String meetingLink,
            String feedback, int score, String result,
            String candidateName, String interviewerName, String jobTitle) {
        this(id, applicationId, interviewerId, interviewDate, type,
                location, meetingLink, feedback, score, result);
        this.candidateName = candidateName;
        this.interviewerName = interviewerName;
        this.jobTitle = jobTitle;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getApplicationId() {
        return applicationId;
    }

    public void setApplicationId(int applicationId) {
        this.applicationId = applicationId;
    }

    public int getInterviewerId() {
        return interviewerId;
    }

    public void setInterviewerId(int interviewerId) {
        this.interviewerId = interviewerId;
    }

    public LocalDateTime getInterviewDate() {
        return interviewDate;
    }

    public void setInterviewDate(LocalDateTime interviewDate) {
        this.interviewDate = interviewDate;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getMeetingLink() {
        return meetingLink;
    }

    public void setMeetingLink(String meetingLink) {
        this.meetingLink = meetingLink;
    }

    public String getFeedback() {
        return feedback;
    }

    public void setFeedback(String feedback) {
        this.feedback = feedback;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public String getResult() {
        return result;
    }

    public void setResult(String result) {
        this.result = result;
    }

    public String getCandidateName() {
        return candidateName;
    }

    public void setCandidateName(String candidateName) {
        this.candidateName = candidateName;
    }

    public String getInterviewerName() {
        return interviewerName;
    }

    public void setInterviewerName(String interviewerName) {
        this.interviewerName = interviewerName;
    }

    public String getJobTitle() {
        return jobTitle;
    }

    public void setJobTitle(String jobTitle) {
        this.jobTitle = jobTitle;
    }

    @Override
    public String toString() {
        return "Interview{id=" + id +
                ", applicationId=" + applicationId +
                ", date=" + interviewDate + "}";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (!(o instanceof Interview))
            return false;
        Interview that = (Interview) o;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/Entity/Recrutement/Interview.php
<?php

namespace App\Entity\Recrutement;

use App\Repository\Recrutement\InterviewRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: InterviewRepository::class)]
#[ORM\Table(name: 'interviews')]
#[ORM\Index(name: 'idx_interviews_date', columns: ['interview_date'])]
#[ORM\Index(name: 'idx_interviews_result', columns: ['result'])]
#[ORM\Index(name: 'idx_interviews_type', columns: ['type'])]
#[ORM\Index(name: 'idx_interviews_is_deleted', columns: ['is_deleted'])]
class Interview
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private int $id;

    #[ORM\ManyToOne(targetEntity: Application::class, inversedBy: 'interviews')]
    #[ORM\JoinColumn(name: 'application_id', referencedColumnName: 'id', nullable: false)]
    #[Assert\NotNull(message: 'Application is required')]
    private ?Application $application = null;

    #[ORM\Column(name: 'interviewer_id', type: Types::INTEGER, nullable: false)]
    #[Assert\NotNull(message: 'Interviewer is required')]
    private ?int $interviewerId = null;

    #[ORM\Column(name: 'interview_date', type: Types::DATETIME_MUTABLE)]
    #[Assert\NotNull(message: 'Interview date is required')]
    #[Assert\GreaterThan('now', message: 'Interview date must be in the future')]
    private ?\DateTimeInterface $interviewDate = null;

    #[ORM\Column(length: 30)]
    #[Assert\NotBlank(message: 'Type is required')]
    #[Assert\Length(max: 30, maxMessage: 'Type cannot exceed 30 characters')]
    private ?string $type = null;

    #[ORM\Column(name: 'meeting_link', length: 500, nullable: true)]
    #[Assert\Length(max: 500, maxMessage: 'Meeting link cannot exceed 500 characters')]
    #[Assert\Url(message: 'Please enter a valid URL')]
    private ?string $meetingLink = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Location cannot exceed 255 characters')]
    private ?string $location = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $feedback = null;

    #[ORM\Column(options: ['default' => 0])]
    #[Assert\GreaterThanOrEqual(value: 0, message: 'Score must be at least 0')]
    #[Assert\LessThanOrEqual(value: 100, message: 'Score cannot exceed 100')]
    private int $score = 0;

    #[ORM\Column(length: 20)]
    #[Assert\NotBlank(message: 'Result is required')]
    #[Assert\Length(max: 20, maxMessage: 'Result cannot exceed 20 characters')]
    private ?string $result = null;

    #[ORM\Column(type: Types::BOOLEAN, options: ['default' => false])]
    private bool $isDeleted = false;

    public function getId(): int
    {
        return $this->id;
    }

    public function getApplication(): ?Application
    {
        return $this->application;
    }

    public function setApplication(?Application $application): static
    {
        $this->application = $application;
        return $this;
    }

    public function getInterviewerId(): ?int
    {
        return $this->interviewerId;
    }

    public function setInterviewerId(?int $interviewerId): static
    {
        $this->interviewerId = $interviewerId;
        return $this;
    }

    public function getInterviewDate(): ?\DateTimeInterface
    {
        return $this->interviewDate;
    }

    public function setInterviewDate(\DateTimeInterface $interviewDate): static
    {
        $this->interviewDate = $interviewDate;
        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;
        return $this;
    }

    public function getMeetingLink(): ?string
    {
        return $this->meetingLink;
    }

    public function setMeetingLink(?string $meetingLink): static
    {
        $this->meetingLink = $meetingLink;
        return $this;
    }

    public function getLocation(): ?string
    {
        return $this->location;
    }

    public function setLocation(?string $location): static
    {
        $this->location = $location;
        return $this;
    }

    public function getFeedback(): ?string
    {
        return $this->feedback;
    }

    public function setFeedback(?string $feedback): static
    {
        $this->feedback = $feedback;
        return $this;
    }

    public function getScore(): ?int
    {
        return $this->score;
    }

    public function setScore(int $score): static
    {
        $this->score = $score;
        return $this;
    }

    public function getResult(): ?string
    {
        return $this->result;
    }

    public function setResult(string $result): static
    {
        $this->result = $result;
        return $this;
    }

    public function isDeleted(): bool
    {
        return $this->isDeleted;
    }

    public function setIsDeleted(bool $isDeleted): static
    {
        $this->isDeleted = $isDeleted;
        return $this;
    }
}
